plat/nvidia/tegra/soc/t186/drivers/smmu/smmu.c

```diff
@@ -438,18 +444,24 @@ void tegra_smmu_save_context(void)
 	assert(!((pgshift != PGSHIFT) || (cb_size != CB_SIZE)));
 #endif
 
+	assert((smmu_ctx_addr >= tzdram_base) && (smmu_ctx_addr <= tzdram_end));
```

Comment:
@vwadekar 
Ahead of merging, can you check that the conditions in this assert are what you intended? If reading correctly, I think that the second check should consider the size of the smmu_ctx_regs array.
Assuming this needs to be corrected, let me know if you want to update this PR or submit a separate PR to address just this?